src/Sulu/Bundle/ContentBundle/Teaser/TeaserContentType.php

```diff
@@ -110,7 +111,32 @@ public function getContentData(PropertyInterface $property)
             return [];
         }
 
-        return $this->teaserManager->find($items, $property->getStructure()->getLanguageCode());
+        $result = $this->teaserManager->find($items, $property->getStructure()->getLanguageCode());
+
+        $mediaReferenceStore = $this->getMediaReferenceStore();
+        if (!$mediaReferenceStore) {
+            return $result;
+        }
+
+        foreach ($result as $item) {
+            if ($item->getMediaId()) {
+                $mediaReferenceStore->add($item->getMediaId());
```

Comment:
Shouldn't also be other stores considered here? AFAIK there could also e.g. pages returned here, right?

Reply:
this is already done in `preResolve` method